bin/apply

```diff
@@ -1,36 +1,25 @@
 #!/bin/bash
 
-#Usage
-USAGE="Usage: kn apply <aws|gce|azure|openstack>"
+# Exit immediately if a command exits with a non-zero status
+set -e
 
-# Validate command
-if [ "$#" -eq 0 ]; then
-  >&2 echo "Error: no arguments supplied"
-  echo "$USAGE"
-  exit 1
-fi
-HOST_CLOUD="$1"
-if [ "$HOST_CLOUD" != 'openstack' ] && \
-  [ "$HOST_CLOUD" != 'gce' ] && \
-  [ "$HOST_CLOUD" != 'azure' ] && \
-  [ "$HOST_CLOUD" != 'aws' ]; then
-  >&2 echo "Error: unrecognized host cloud '$HOST_CLOUD'"
-  echo "$USAGE"
+HOST_CLOUD=$(grep -w "provider" terraform.tfvars | cut -d "=" -f 2- | awk -F\" '{print $(NF-1)}')
+
+# Check for recognized cloud provider
+if ! grep -qw "$HOST_CLOUD" <<< "openstack gce azure aws"; then
+  >&2 echo "Error: unrecognized host cloud '$HOST_CLOUD' in config file terraform.tfvars"
   exit 1
 fi
 
-#Exit immediately if a command exits with a non-zero status
-set -e
-
 # Import image (AWS doesn't need it)
-if [ "$HOST_CLOUD" = 'openstack' ] || [ "$HOST_CLOUD" = 'gce' ]; then
-  ansible-playbook -e "credentials_file_path=$PWD/service-account.json" /opt/KubeNow/playbooks/import-"$HOST_CLOUD"-image.yml
-elif [ "$HOST_CLOUD" = 'azure' ]; then
-  /opt/KubeNow/bin/image-create-azure.sh
+if [ "$HOST_CLOUD" = 'openstack' ] || [ "$HOST_CLOUD" = 'gce' ] || [ "$HOST_CLOUD" = 'gce' ]; then
```

Comment:
There is HOST_CLOUD gce twice, maybe you mean azure

Reply:
yes:-) fixed